src/odemis/driver/tfsbc.py

```diff
@@ -209,8 +219,7 @@ def _setShift(self, value):
         self._write_registers(register_values)
 
         # Scale back to multiprobe units and convert back to multiprobe coordinates
-        value = (value[0] / gain[0], value[1] / gain[1])
-        value = transform_coordinates(value, -rotation, (-offset[0], -offset[1]), LH, RH)
+        value = transform_coordinates(value, rotation, numpy.array(gain), offset, LH, RH, inverse=True)
```

Comment:
I have a hard time to convince myself it's not an overkill. You call this function twice, once with arguments RH,LH,inverse=False and once with LH,RH,inverse=True. It feels that the inverse is only doing half of its work.
Moreover, assuming that the rotation,gain, offset have been specifically computed for this component, I don't see why you'd need to switch between RH and LH. Isn't it like having a gain with a -Y? 
Also, I feel that your original idea of using SimilarityTransform(rotation,gain,offset) should work. Did you try calling `.inverse()` of this transformation, and this new transform didn't allow you to get the original values? That'd be a bug in that code then.

Reply:
**Scaling Transform**
It turned out that the DC coordinate system is not completely orthogonal. The angle between the x and y axes is less than 90 degrees. This means we need a nonlinear transform. There is a plugin that calculates this transform in the external software. The goal is now to use this transform. This means that I have to change the driver code again. I don't think that the alignment code is necessary anymore.
Probably this nonlinearity was the reason that the ScalingTransform was not working. Doing the rotation + scaling manually gave much better results for some reason (not sure why though, the parameters of the ScalingTransform were the same). I don't think there is a bug.

**transform_parameters**
Why do you think the inverse is doing half of the work? The main problem is, that the order of the scaling and rotation matters. We always scale in the dc frame, so for the inverse we need to do scaling first, then rotation, for the regular transformation it's the other way around.
We need to switch between RH and LH because the value which is given to the driver is in RH, but the dc hardware operates in LH.

Reply:
**Scaling Transform**
So let's see, hopefully this transform will really be available soon. BTW, I don't get how "manually" you can call ScalingTransform with the same parameters and get different results. The transform is very precisely defined mathematically. There is no reason for a computer to interpret it differently than you.

**transform_parameters**
The way that Transform classes are defined is such that if you use the inverse() version, it directly works. The rotation/scaling order is taken care by the inverse() computation.
Concerning the RH/LH, I understand that they are different, but I don't understand why you have it to do so. My feeling is that the calibration is specifically computed for a given hardware, so the numbers should directly fit. To switch between LH and RH, isn't it just a matter of putting a negative sign on the Y scale? I think it'd be better that the calibration algorithm knows (or detects) that an axis is inverted, instead of hard-coding it in the driver.